scripts/Tools/compare_test_results

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+"""
+Analyze results from a test root area, comparing non-BFB
+changes. Purpose is, instead of re-running tests
+in compare mode, which is very slow, allow for very fast analsis of diffs.
+
+You may need to load modules for cprnc to work.
+"""
+
+from standard_script_setup import *
+
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.compare_test_results import compare_test_results
+
+import argparse, sys, os, doctest
+
+_MACHINE = Machines()
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+usage="""\n%s [-n] [-r <TESTROOT>] [-b <BRANCH>] [-c <COMPILER>] [<TEST> <TEST> ...] [--verbose]
+OR
+%s --help
+OR
+%s --test
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# From most recent run, compare all changes \033[0m
+    > %s
+    \033[1;32m# From most recent run, compare only changes for test foo and bar only \033[0m
+    > %s -n foo bar
+    \033[1;32m# From most recent run of jenkins, compare history changes for next \033[0m
+    > %s -r /home/jenkins/acme/scratch/jenkins -b next
+""" % ((os.path.basename(args[0]), ) * 6),
+
+description=description,
+
+formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+    default_baseline_name = CIME.utils.get_current_branch(repo=CIME.utils.get_cime_root())
```

Comment:
Replying to a few related comments here:

> It's not used for much. It allows the user to specify a baseline_name + compiler as a way of selecting tests instead of the test_id.

> The default baseline dir is: baseline_root/compiler/branch_name, so it's not unreasonable to use the current branch as the default.

> This had to do with identifying a set of tests via test_id or baseline_name + compiler that I mentioned earlier. It is convenient for some scripts not to have to know the test_id.

Ah, perhaps this is a difference in the ACME vs. CESM workflows??? For CESM, the baseline directory is typically a frozen tag, not the name of an evolving branch. If these defaults fit in with the ACME workflow, I'm okay with them being kept as is. Maybe the real need then is for explicit documentation of the expected usage with the CESM workflow... I _think_ that, for the CESM workflow, you would specify -b and -t but not -c; do you agree? Also, it would help to document that the --compiler option is only used if --test-id is not specified (I think).